### PR TITLE
Task reduce bundle size

### PR DESCRIPTION
**Pull request by timvancleef**
# Pull Request Template

## Description

This PR reduces our bundle size by:

* removing (currently) unused fontawesome package
* changing usage of `buefy` to only include used components (https://buefy.org/documentation/start#npm-or-yarn: Individual components as Vue plugins)
* using `@aws-amplify` packages instead of global `aws-amplify` module and explicitly only using graphql-api since we dont need the rest-api.

This reduces the size of our `vendor.js` from 2.02mb parsed size down to 617kb and the overall bundle size from sth. like 4mb down to 1mb.

Those sizes are still large, but it is something 😄... i think most of the size comes from `@aws-amplify` at least regarding our bundle report.
Bildschirmfoto 2021-04-30 um 12 09 54

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

I used the `bundle-analyzer` which is bundled with `vue-cli` to compare sizes before and after the changes:
report_old
report

I made sure that `npm run serve` as well as `npm run build && npx http-server dist` still work with my deployt backend by:

- logging in and out of the app
- reporting negative and positive test results

Didnt notice any crashes or error logs.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by cremich** [comment]
@timvancleef what kind of documentation update do you see after we merge this? Referring to this because you checked "This change requires a documentation update"

**Comment by timvancleef** [comment]
> 
> 
> @timvancleef what kind of documentation update do you see after we merge this? Referring to this because you checked "This change requires a documentation update"

Ah, maybe i misunderstood that checkmark. I added the `analyze` npm task, and therefore i also updated the readme.
This is what i meant with "requires" documentation update.

**Review by cremich — DISMISSED** [review]

**Comment by cremich** [comment]
@timvancleef Damn...I tried to resolve the conflicts after I merged the coverage PR. Will fix this here during the day :)

File: package.json [hunk 1]
```diff
@@ -4,16 +4,21 @@
   "private": true,
   "scripts": {
     "serve": "vue-cli-service serve",
-    "build": "vue-cli-service build",
+    "build": "vue-cli-service build --modern",
     "test": "npm run test:unit",
     "test:coverage": "npm run test:unit -- --coverage",
     "test:unit": "vue-cli-service test:unit",
     "lint": "vue-cli-service lint",
+    "analyze": "npm run build -- --report"
```
Comment (timvancleef) [inline package.json]:
```suggestion
    "analyze": "npm run build -- --report",
```

**Comment by timvancleef** [comment]
> @timvancleef Damn...I tried to resolve the conflicts after I merged the coverage PR. Will fix this here during the day :)

I think it was just a typo in package.json. Should be fixed 🙂

**Review by cremich — APPROVED** [review]

**Pull request merged**